### PR TITLE
Allow installing both port:tbb and port:onetbb

### PR DESCRIPTION
**Pull request by catap**
#### Description

Here a PR on which I've worked for last weeks.

The goal is allow to install both `tbb` and `onetbb` port on the same machine, and it forces to add installing both `oce` and `opencascade`.

It is based on serveral PRs which I've made and which's fixed broken ports:
 - https://github.com/macports/macports-ports/pull/16866
 - https://github.com/macports/macports-ports/pull/16876
 - https://github.com/macports/macports-ports/pull/16888
 - https://github.com/macports/macports-ports/pull/16893
 - https://github.com/macports/macports-ports/pull/16899
 - https://github.com/macports/macports-ports/pull/16912
 - https://github.com/macports/macports-ports/pull/18388

To be clear: only the last 3 commits are new.

Thus, I've skip CI because it had no chance to rebuild everything in 6 hours.

###### Type(s)

- [x] bugfix
- [x] enhancement

###### Tested on

macOS 12.6 21G115 x86_64
Xcode 14.1 14B47b

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: cad/freecad/Portfile [hunk 1]
```diff
@@ -6,9 +6,10 @@ PortGroup               compilers 1.0
 PortGroup               github 1.0
 PortGroup               qt4 1.0
 PortGroup               boost 1.0
+PortGroup               active_variants 1.1
 
 github.setup            FreeCAD FreeCAD 0.18.5
-revision                1
+revision                3
```
Comment (mascguy) [inline cad/freecad/Portfile:12]: Should the new revision be 2 instead?

File: graphics/blender/Portfile [hunk 2]
```diff
@@ -7,8 +7,8 @@ PortGroup           active_variants 1.1
 PortGroup           boost 1.0
 
 name                blender
-version             2.91.2
-revision            6
+version             2.93.11
+revision            1
```
Comment (mascguy) [inline graphics/blender/Portfile:11]: Should the revision be zero?

File: math/deal.ii/Portfile [hunk 3]
```diff
@@ -7,9 +7,9 @@ PortGroup               mpi            1.0
 PortGroup               linear_algebra 1.0
 PortGroup               boost          1.0
 
-github.setup            dealii dealii 9.3.1 v
+github.setup            dealii dealii 9.4.0 v
+revision                1
```
Comment (mascguy) [inline math/deal.ii/Portfile:11]: Reset to zero

File: science/gmsh/Portfile [hunk 4]
```diff
@@ -7,8 +7,8 @@ PortGroup               linear_algebra 1.0
 PortGroup               muniversal     1.0
 
 name                    gmsh
-version                 4.9.3
-revision                2
+version                 4.11.0
+revision                1
```
Comment (mascguy) [inline science/gmsh/Portfile:11]: Reset to zero

File: science/kicad/Portfile [hunk 5]
```diff
@@ -29,7 +29,7 @@ if {${name} eq ${subport}} {
     checksums           rmd160  3603cc21857468ed0edb8fe941c92776004237eb \
                         sha256  808063f728a8c3ed42af2126b465c9a4aa3e6df8e025387b9b11c6a15710d9ce \
                         size    34045371
-    revision            0
+    revision            2
```
Comment (mascguy) [inline science/kicad/Portfile:32]: Should this be 1 instead?

File: science/nektarpp/Portfile [hunk 6]
```diff
@@ -12,7 +12,7 @@ gitlab.setup            nektar nektar 5.2.0 v
 boost.version           1.71
 
 name                    nektarpp
-revision                1
+revision                3
```
Comment (mascguy) [inline science/nektarpp/Portfile:15]: Should this be 2 instead?

File: science/opencascade/Portfile [hunk 7]
```diff
@@ -5,8 +5,8 @@ PortGroup                   cmake      1.1
 PortGroup                   muniversal 1.0
 
 name                        opencascade
-version                     7.6.0
-revision                    0
+version                     7.7.0
+revision                    1
```
Comment (mascguy) [inline science/opencascade/Portfile:9]: Reset to zero

File: devel/tbb/Portfile [hunk 8]
```diff
@@ -35,6 +31,9 @@ use_configure       no
 # Force a compatible compiler
 compiler.blacklist-append *gcc* {clang < 602} {macports-clang-3.[0-7]}
 
+# install it into libexec to avoid conflict with onetbb port
+prefix              ${prefix}/libexec/${name}
```
Comment (mascguy) [inline devel/tbb/Portfile:35]: AFAIK, we typically don't change `prefix` within a port, as that could cause unexpected issues.

Instead, it's better to define your own variable, representing the install prefix for your port.

Anyone else... thoughts?

File: devel/tbb/Portfile [hunk 9]
```diff
@@ -97,7 +96,14 @@ destroot {
...
+    xinstall -d ${destroot}${prefix}/share/
+    copy ${worksrcpath}/doc ${destroot}${prefix}/share/doc
+    copy ${worksrcpath}/examples ${destroot}${prefix}/share/examples
 }
+
+livecheck.type      none
+
+notes "
+More recent versions are released as oneTBB and are packaged in the new 'onetbb' port
+This version is installed into ${prefix}/libexec/${name}
```
Comment (mascguy) [inline devel/tbb/Portfile:108]: Replace hard-coded path with your new variable, per prior comment.

**Comment by mascguy** [comment]
Very happy to see someone segregate the TBB ports, allowing them to finally co-exist. Ditto for updating the various ports.

Great stuff Kirill!

**Comment by mascguy** [comment]
I did a quick high-level review of the various Portfile changes, and added comments. But no serious concerns overall.

Anyone else?

**Comment by catap** [comment]
@mascguy thanks for review. Almost all commits from this PRs is parts from another ones which I've mentored inside the first commit.

It explains "wired" bump of revisions ;)

**Comment by catap** [comment]
All dependencies of this PR have been merged. So, it is ready.

@mascguy may I ask you to reivew it?

**Review by mascguy — APPROVED** [review]

**Comment by catap** [comment]
Yahoo! Thanks @mascguy!

**Comment by catap** [comment]
I guess it was one of the longest PR that I've done so far :)

**Comment by mascguy** [comment]
> I guess it was one of the longest PR that I've done so far :)

Great work as always Kirill, thank you for everything you do!

**Comment by mohd-akram** [comment]
I noticed this moved onetbb to libexec as well. Is this necessary since tbb is already moved there?

**Comment by mascguy** [comment]
> I noticed this moved onetbb to libexec as well. Is this necessary since tbb is already moved there?

Yes, we need to move all versions to `libexec`, otherwise ports may inadvertently pick up `onetbb` when they want `tbb`.

**Comment by catap** [comment]
> I noticed this moved onetbb to libexec as well. Is this necessary since tbb is already moved there?

and some magic in cmake level allows to not touch any port which is used OneTBB.

Thus, now we have ability to keep old version OneTBB for ports which for some reason need it.

**Comment by mascguy** [comment]
Kirill, I noticed one potential issue: It looks like the CMake files are being installed in two places, for `onetbb`. The segregated versions under libexec are fine, but they're also being placed in `${prefix}/lib/cmake`:

```
Port onetbb contains:
  /opt/local/lib/cmake/TBB/TBBConfig.cmake
  /opt/local/lib/cmake/TBB/TBBConfigVersion.cmake
  /opt/local/lib/cmake/TBB/TBBTargets-macports.cmake
  /opt/local/lib/cmake/TBB/TBBTargets.cmake
[...]
  /opt/local/libexec/onetbb/lib/cmake/TBB/TBBConfig.cmake
  /opt/local/libexec/onetbb/lib/cmake/TBB/TBBConfigVersion.cmake
  /opt/local/libexec/onetbb/lib/cmake/TBB/TBBTargets-macports.cmake
  /opt/local/libexec/onetbb/lib/cmake/TBB/TBBTargets.cmake
```

**Comment by catap** [comment]
@mascguy it isn't an issue. It was made for propose. Why? To avoid supply each port with the path to the right OneTBB.

`OneTBB` is succesor of `TBB` and everyone should switch to it. And if you cat, for example, `/opt/local/lib/cmake/TBB/TBBConfig.cmake` you may see that it simple says:
```
include(/opt/local/libexec/onetbb/lib/cmake/TBB/TBBTargets.cmake)
```

:)

This migration is transparant for any port which is compatible with OneTBB and uses CMake. I assume it is majority of ports.

**Pull request merged**